scripts/utils.sh

```diff
@@ -147,3 +147,22 @@ function get_ipv4_address {
     echo "${hipv4}"
     return 0
 }
+
+function verify_nvidia-smi {
```

Comment:
I think the function name is misleading. What is this really doing? It is checking if there is an nvidia gpu is available. nvidia-smi is the tool we use to do that, but it is not the important thing. I think a better name is `nvidia_gpu_available`, and either it is or it isn't (no need for anything other than 0 or 1)